### PR TITLE
mac adjustment for playsound

### PR DESCRIPTION
**Pull request by Decksname**

### CODE REVIEW

File: nipple.py [hunk 1]
```diff
@@ -13,6 +15,12 @@
 
 force_playsound = len(os.sys.argv) > 1 and os.sys.argv[1] == '--playsound'
 
+# Define if nipple runs on a macOS system
+isMac = False
+if platform.system() == "Darwin":
+    isMac = True
+
```
Comment (BuZZ-T) [inline nipple.py]: isMac = platform.system() == "Darwin"

File: nipple.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env python
 
 import os
+import platform
+# import os.path
```
Comment (BuZZ-T) [inline nipple.py]: Please remove comment

File: nipple.py [hunk 3]
```diff
@@ -29,27 +37,30 @@
...
-        try:
-            global current_sound_file
-            pygame.mixer.music.stop()
-            if not sound_file is current_sound_file:
-                current_sound_file = sound_file
-                pygame.mixer.music.load(sound_file)
-            pygame.mixer.music.play()
-        except pygame.error:
-            logger.debug('no file "%s" found' % sound_file)
+    if os.path.isfile(sound_file):
```
Comment (BuZZ-T) [inline nipple.py]: the `except pygame.error` and `except PlaysoundException` are handling this case, so this shouldn't be necessary

**Review by BuZZ-T — COMMENTED** [review]
Just a tip: It's better to create a feature branch on your fork and create the pull request from this branch to the original repository

File: nipple.py [hunk 4]
```diff
@@ -61,9 +72,15 @@ def on_press(key):
         shift_pressed = True
     elif str(key)[:5] == 'Key.f':
         if ctrl_pressed and shift_pressed:
-            play('nipple_shift_%s.ogg' % str(key)[4:])
+            if isMac:
+                play('nipple_shift_%s.mp3' % str(key)[4:])
```
Comment (BuZZ-T) [inline nipple.py:68]: Can you add a global variable FILE_EXTENSION and set it to "mp3" or "ogg" and call it here via `'nipple_shift_%s.%s'  % (str(key)[4:], FILE_EXTENSION)` instead?

File: nipple.py (hunk 1 above)
Comment (BuZZ-T) [reply nipple.py]: outdated, replace with `FILE_EXTENSION = 'mp3' if platform.system() == 'Darwin' else 'ogg'`

File: nipple.py (hunk 2 above)
Comment (Decksname) [reply nipple.py]: done

File: nipple.py (hunk 1 above)
Comment (Decksname) [reply nipple.py]: done. tnx.

File: nipple.py (hunk 3 above)
Comment (Decksname) [reply nipple.py]: Unfortunately it is necessary. The playsound implementation for mac crashes when the file does't exist.

**Review by BuZZ-T — APPROVED** [review]

**Pull request merged**